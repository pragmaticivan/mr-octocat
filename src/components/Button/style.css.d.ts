export const btn_follow: string;
export const btnFollow: string;
