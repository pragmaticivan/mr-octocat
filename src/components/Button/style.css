.btn_follow {
  width: 100%;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  padding: 6px 12px;
  white-space: nowrap;
  cursor: pointer;
  border-radius: 6px;
  border: 1px solid rgba(27,31,35,.2);
  background-color: #eff3f6;
  background-image: linear-gradient(-180deg,#fafbfc,#eff3f6 90%);
  color: #24292e;
}

.btn_follow:hover {
  background-color: #e6ebf1;
  background-image: linear-gradient(-180deg,#f0f3f6,#e6ebf1 90%);
  background-position: -.5em;
  border-color: rgba(27,31,35,.35);
}

.btn_follow:active {
  background-color: #e9ecef;
  background-image: none;
  border-color: rgba(27,31,35,.35);
  box-shadow: inset 0 0.15em 0.3em rgba(27,31,35,.15);
}

