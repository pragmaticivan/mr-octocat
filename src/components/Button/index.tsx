import React from "react";
import style from "./style.css";

interface Props {
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const Button: React.FC<Props> = (props) => {
  const { onClick, children } = props;
  return (
    <button className={style.btn} onClick={onClick}>
      {children}
    </button>
  );
}

export default Button;
