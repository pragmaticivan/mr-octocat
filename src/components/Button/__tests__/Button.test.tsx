import Button from '../';
import React from 'react';
import { shallow } from 'enzyme';

describe('Button', () => {
  it('renders the component', () => {
    const wrapper = shallow(<Button>Test BTN</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children', () => {
    const wrapper = shallow(<Button>Test BTN</Button>);

    expect(wrapper.text()).toEqual('Test BTN');
  });
});
