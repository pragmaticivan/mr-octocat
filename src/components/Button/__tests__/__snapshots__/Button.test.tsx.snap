// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component 1`] = `ShallowWrapper {}`;
